pkg/network/setup/controller_test.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package network_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "kubevirt.io/api/core/v1"
+	netsetup "kubevirt.io/kubevirt/pkg/network/setup"
+)
+
+var _ = Describe("controller", func() {
+	var netCtrl netsetup.Controller
+	var vmi *v1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		netCtrl = netsetup.NewController(nil)
+
+		vmi = &v1.VirtualMachineInstance{}
+		vmi.UID = "123"
+	})
+
+	It("runs setup successfully", func() {
+		Expect(netCtrl.Setup(vmi, 0, doNetNSDummyNoop, netPreSetupDummyNoop)).To(Succeed())
+		Expect(netCtrl.SetupCompleted(vmi)).To(BeTrue())
+	})
+
+	It("runs teardown successfully", func() {
+		Expect(netCtrl.Setup(vmi, 0, doNetNSDummyNoop, netPreSetupDummyNoop)).To(Succeed())
+		Expect(netCtrl.SetupCompleted(vmi)).To(BeTrue())
```

Comment:
In teardown contexts, I see this as part of the setup stage.

Would you be open to have this in a separate context, and move this into a before each block ? I think that helps the reader.

Reply:
I disagree, and while it does require boilerplate, it brings structure to the code that helps me figure out how the test works (or focus exactly in the relevant part of it).

Anyway, this is just an opinionated suggestion: I personally prefer to have the code structured like that, but, I can't enforce it.